tests/compact_gate.rs

```diff
@@ -0,0 +1,10 @@
+use std::process::Command;
+
+#[test]
+fn compact_gate() {
```

Comment:
this takes ~1m on my machine each time I run `cargo test`. That is probably too slow for a test that runs that often. We should push most of them to the CI and run just a couple locally

Reply:
Did it run in the `pre-commit`? I thought I had this only in CI and not pre-commit.

Reply:
I think I saw this by just running `cargo test`